lib/Doctrine/MongoDB/EagerCursor.php

```diff
@@ -107,7 +121,7 @@ public function getSingleResult()
     public function initialize()
     {
         if ($this->initialized === false) {
-            $this->data = $this->cursor->toArray();
+            $this->data = $this->cursor->toArray($this->useKeys);
```

Comment:
Although we discussed moving `$useKeys` to an `initialize()` parameter, integration with ODM would benefit from keeping this as a constructor argument. That is because ODM's Cursor class (if we remove its EagerCursor) will merely wrap a CursorInterface.